src/rebar_prv_eunit.erl

```diff
@@ -134,17 +134,28 @@ resolve_tests(State) ->
     Files        = resolve(file, RawOpts),
     Modules      = resolve(module, RawOpts),
     Suites       = resolve(suite, module, RawOpts),
-    Apps ++ Applications ++ Dirs ++ Files ++ Modules ++ Suites.
+    Generator    = resolve(generator, RawOpts),
+    Apps ++ Applications ++ Dirs ++ Files ++ Modules ++ Suites ++ Generator.
 
 resolve(Flag, RawOpts) -> resolve(Flag, Flag, RawOpts).
 
 resolve(Flag, EUnitKey, RawOpts) ->
     case proplists:get_value(Flag, RawOpts) of
         undefined -> [];
-        Args      -> lists:map(fun(Arg) -> normalize(EUnitKey, Arg) end,
-                               rebar_string:lexemes(Args, [$,]))
+        Args      ->
+            lists:flatten(lists:map(fun(Arg) -> normalize(EUnitKey, Arg) end,
+                                    rebar_string:lexemes(Args, [$,])))
     end.
 
+normalize(generator, Value) ->
+    case string:tokens(Value, [$:]) of
+        [Module0, Functions] ->
+            Module = list_to_atom(Module0),
+            lists:map(fun(F) -> {generator, Module, list_to_atom(F)} end,
+                      string:tokens(Functions, [$;]));
+        _ ->
+            ?PRV_ERROR({generator, {"Generator `~p` is invalid format", {Value}}})
```

Comment:
The error should be `{generator, Value}` and move `"Generator ~p has an invalid format"` to a pattern that matches `{generator, Val}` in the  `format_error` callback.

The way to see it is that we could, for example, configure `format_error` to care about languages and locales. If the error message itself is hardcoded in English in the call-site, we lose all flexibility to modify or change how they are handled :)

Reply:
Oh, I got it.
The new pattern is added to `format_error/1` callback.
Thanks for your kindly description. :smile: